src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,220 @@
+import $ from 'jquery';
+import {orderToggleState} from '../../enums/searchToggleState';
+
+export class CourtsTableSearch {
+
+  private static toggleClosedCourtsDisplay = 'toggleClosedCourtsDisplay';
+  private static numberOfCourts = '#numberOfCourts';
+  private static searchCourtsFilterId = '#searchCourts';
+  private static courtsHiddenId = 'courtsHidden';
+  private static courtsNameAscToggleId = '#courtsNameAscToggle';
+  private static courtsUpdatedAscToggleId = '#courtsUpdatedAscToggle';
+  private static courtsResultsSection = '#courtResults > tbody';
+  private static courtsTableHeaderAsc = 'courts-table-header-asc';
+  private static courtsTableHeaderDesc = 'courts-table-header-desc';
+  private static courtsTableHeaderInactive = 'courts-table-header-inactive';
+  private static courtsFrontendUrl = '#courtsFrontendUrl';
+  private static tableCourtsNameId = '#tableCourtsName';
+  private static tableCourtsUpdatedId = '#tableCourtsUpdated';
+
+  /**
+   *
+   * Sets up the table view for the main page, on load the default configuration
+   * is used, otherwise it is updated on each search toggle or courts search
+   *
+   * @param filterName Either 'name' or 'date' depending on the type of toggle
+   * @param defaultValue set to false by default on initial load
+   */
+  static setUpTable(filterName: string, defaultValue = false): void {
+    defaultValue
+      ? CourtsTableSearch.setUpTableData(
+        (document.getElementById(this.courtsHiddenId) as HTMLInputElement).value,
+      $(this.searchCourtsFilterId).val() as string, // if the user clicks 'back' in the browser after view or editing
+      false, orderToggleState.ASC, orderToggleState.INACTIVE)
+      : CourtsTableSearch.setUpTableData((
+        document.getElementById(this.courtsHiddenId) as HTMLInputElement).value,
+        $(this.searchCourtsFilterId).val() as string,
+        $(`#main-content input[name=${this.toggleClosedCourtsDisplay}]`).prop('checked'),
+        $(this.courtsNameAscToggleId).val() as string,
```

Comment:
Do these need to be hidden inputs? Could these be variables with an initial value?

Reply:
I think it does need to stay as a hidden input, as it needs to toggle between the state each time the relevant column header is clicked on